### PR TITLE
Fix compiler warning in 32bits due to size mismatch in ptr cast

### PR DESCRIPTION
**Pull request by pmatos**
Casting a ptr to an integer of a different size causes a warning -
this is only noticeable on 32bits. Fix this by casting the ptr first
to uintptr_t.

### CODE REVIEW

File: racket/src/ChezScheme/c/gcwrapper.c [hunk 1]
```diff
@@ -576,7 +578,7 @@ static void check_pointer(ptr *pp, IBOOL address_is_meaningful, ptr base, uptr s
       }
 
       if (address_is_meaningful) {
-        seginfo *ppsi = MaybeSegInfo(ptr_get_segment(pp));
+        seginfo *ppsi = MaybeSegInfo(ptr_get_segment((uintptr_t)pp));
```
Comment (mflatt) [inline racket/src/ChezScheme/c/gcwrapper.c:581]: The `TO_PTR` macro should be used here: `TO_PTR(pp)`.

File: racket/src/ChezScheme/c/gcwrapper.c [hunk 2]
```diff
@@ -854,7 +856,7 @@ void S_check_heap(aftergc, mcg) IBOOL aftergc; IGEN mcg; {
                     pp1 = TO_VOIDP((ptr)((uptr)TO_PTR(pp1) + size_object(p)));
                   } else {
                     /* skip past unmarked */
-                    pp1 = TO_VOIDP((ptr)((uptr)pp1 + byte_alignment));
+                    pp1 = TO_VOIDP((ptr)((uintptr_t)pp1 + byte_alignment));
```
Comment (mflatt) [inline racket/src/ChezScheme/c/gcwrapper.c:859]: This should be `(uptr)TO_PTR(pp1)` in place of the original `(uptr)pp1`.

**Review by mflatt — CHANGES_REQUESTED** [review]
In case I miss one going the other direction in the future, the `TO_VOIDP` macro goes from a `ptr` to a `void*`.

Thanks!

**Comment by pmatos** [comment]
Thanks @mflatt 
That makes sense. I will make the changes.

**Review by mflatt — APPROVED** [review]

**Pull request merged**